From 9e8f6c61e1cd05bc211297c1edcadecfa20779c6 Mon Sep 17 00:00:00 2001
From: Robert Adams <misterblue@misterblue.com>
Date: Wed, 14 Dec 2022 04:23:51 +0000
Subject: [PATCH] Add island sleeping transform event

---
 src/BulletDynamics/Dynamics/btDiscreteDynamicsWorld.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/BulletDynamics/Dynamics/btDiscreteDynamicsWorld.cpp b/src/BulletDynamics/Dynamics/btDiscreteDynamicsWorld.cpp
index 9e99c154f..71af633b4 100644
--- a/src/BulletDynamics/Dynamics/btDiscreteDynamicsWorld.cpp
+++ b/src/BulletDynamics/Dynamics/btDiscreteDynamicsWorld.cpp
@@ -616,6 +616,9 @@ void btDiscreteDynamicsWorld::updateActivationState(btScalar timeStep)
 					{
 						body->setAngularVelocity(btVector3(0, 0, 0));
 						body->setLinearVelocity(btVector3(0, 0, 0));
+                        // when sleeping, force the motion state to be called/updated
+                        if (body->getMotionState())
+                            body->getMotionState()->setWorldTransform(body->getWorldTransform());
 					}
 				}
 			}
-- 
2.25.1

